### ISSUE TITLE
Better documentation for transitive header and transitive libs (traits)

### ISSUE THREAD

**Issue by uilianries** [conan-io/docs#3456]
When looking for `transitive_headers` and `transitive_libs` parameter for [requirements](https://docs.conan.io/2/reference/conanfile/methods/requirements.html#requirements), I can find:

- https://docs.conan.io/2/reference/conanfile/methods/requirements.html#transitive-headers
- https://docs.conan.io/2/reference/conanfile/methods/requirements.html#transitive-libs

Still, the documentation only specifies the WHAT, not the WHY, or recommended cases. 

Right now, the [requirements traits](https://github.com/conan-io/tribe/blob/main/design/026-requirements_traits.md) design is more informative, and could be used as inspiration.

It would be great if we have a better explanation, like:

- Transitive headers are useful when a public header of a project has included a downstream header, resulting that the consumer will need to have access to both upstream and downstream headers

**Comment by valgur** [comment]
Yes, please. I wish the docs also mentioned how the default values for these two are determined - that they are mostly affected by the `package_type` attribute (and also maybe how) and the `libs` and `headers` parameters. The fact that `transitive_libs` is not needed for `header-library` but is for others can be a bit surprising at first.
